homeassistant/components/esphome/entry_data.py

```diff
@@ -57,6 +70,23 @@
     TextSensorInfo: "sensor",
 }
 
+STATE_TYPE_TO_COMPONENT_KEY = {
+    BinarySensorState: "binary_sensor",
```

Comment:
Let's use the `Platform` enum https://github.com/home-assistant/core/blob/dev/homeassistant/const.py#L22

Reply:
Adjusting....